app/src/main/java/com/example/farmeraid/data/QuotasRepository.kt

```diff
@@ -1,115 +1,78 @@
 class QuotasRepository {
     data class ProduceQuota(
-        val produceName : String,
-        val produceGoalAmount : Int,
+        val produceName: String,
+        val produceGoalAmount: Int,
+        val saleAmount: Int
     )
 
     data class Quota(
         val id: String,
-        val produceQuotaList : List<ProduceQuota>,
+        val produceQuotaList: List<ProduceQuota>
     )
 
-    private var currentId : String = "5"
+    private var db: FirebaseFirestore = FirebaseFirestore.getInstance()
 
-    private val quotasList: MutableList<Quota> = mutableListOf(Quota(
-        id = "1",
-        produceQuotaList = listOf(
-            ProduceQuota(
-                produceName = "Apple",
-                produceGoalAmount = 25,
-            ),
-            ProduceQuota(
-                produceName = "Banana",
-                produceGoalAmount = 10,
-            ),
-            ProduceQuota(
-                produceName = "Mango",
-                produceGoalAmount = 1,
-            ),
-            ProduceQuota(
-                produceName = "Strawberry",
-                produceGoalAmount = 2,
-            ),
-        )
-    ), Quota(
-        id = "2",
-        produceQuotaList = listOf(
-            ProduceQuota(
-                produceName = "Strawberry",
-                produceGoalAmount = 15,
-            ),
-            ProduceQuota(
-                produceName = "Apple",
-                produceGoalAmount = 24,
-            ),
-        )
-    ), Quota(
-        id = "3",
-        produceQuotaList = listOf(
-            ProduceQuota(
-                produceName = "Apple",
-                produceGoalAmount = 4,
-            ),
-            ProduceQuota(
-                produceName = "Banana",
-                produceGoalAmount = 10,
-            ),
-            ProduceQuota(
-                produceName = "Mango",
-                produceGoalAmount = 5,
-            ),
-        )
-    ), Quota(
-        id = "4",
-        produceQuotaList = listOf(
-            ProduceQuota(
-                produceName = "Banana",
-                produceGoalAmount = 5,
-            ),
-            ProduceQuota(
-                produceName = "Mango",
-                produceGoalAmount = 12,
-            ),
-        )
-    ),
+    //Todo: Refactor to not mock sales data
+    private var SALES = hashMapOf(
+        "mangoes" to 4,
+        "apple" to 2,
+        "strawberry" to 7,
+        "bananas" to 2,
+        "strawberry" to 4
     )
 
-    fun getCategorizedQuotas(): Flow<MutableList<Quota>> {
-        return flow {
-            emit(quotasList)
-        }.flowOn(Dispatchers.IO)
-    }
+    suspend fun getQuota(id: String): Quota? {
+        val docRef = db.collection("quotas").document(id)
 
-    fun getQuota(id : String): Quota? {
-        return quotasList.firstOrNull { quota -> quota.id == id }
-    }
+        val quotas = docRef.get().await()?.data?.get("produce")
+        val saleCount = docRef.get().await()?.data?.get("sale")
```

Comment:
do docRef.get().await()? once and store it in a variable. You're doing two document reads here.

Reply:
I need to pull both dicts right? One for sale and one for produce

Reply:
This is how the collection looks like:
![image](https://github.com/adshayanB/ECE452/assets/41243143/9bc8c3dc-7cbc-4fd6-813b-8eedc8c5fa49)